src/commands/test.js

```diff
@@ -0,0 +1,498 @@
+      --skip-wait-for-ethereum  Don't wait for Ethereum to be up at localhost:8545 (optional)
+      --skip-wait-for-postgres  Don't wait for Postgres to be up at localhost:5432 (optional)
+`
+
+module.exports = {
+  description: 'Tests the current project against a Graph Node and Parity testnet',
+  run: async toolbox => {
+    // Obtain tools
+    let { filesystem, print, system } = toolbox
+
+    // Parse CLI parameters
+    let {
+      composeFile,
+      h,
+      help,
+      nodeImage,
+      nodeLogs,
+      skipWaitForEthereum,
+      skipWaitForIpfs,
+      skipWaitForPostgres,
+      standaloneNode,
+      standaloneNodeArgs,
+    } = toolbox.parameters.options
+
+    // Support both short and long option variants
+    help = help || h
+
+    // Extract test command
+    let params = fixParameters(toolbox.parameters, {
+      h,
+      help,
+      nodeLogs,
+      skipWaitForEthereum,
+      skipWaitForIpfs,
+      skipWaitForPostgres,
+    })
+
+    // Show help text if requested
+    if (help) {
+      print.info(HELP)
+      return
+    }
+
+    if (params.length == 0) {
+      print.error(`Test command not provided as the last argument`)
+      process.exitCode = 1
+      return
+    }
+
+    let testCommand = params[0]
+
+    // Create Docker client
+    let docker = new Docker()
+
+    // Obtain the Docker Compose file for services that the tests run against
+    composeFile =
+      composeFile ||
+      path.join(
+        __dirname,
+        '..',
+        '..',
+        'resources',
+        'test',
+        standaloneNode ? 'docker-compose-standalone-node.yml' : 'docker-compose.yml',
+      )
+    if (!filesystem.exists(composeFile)) {
+      print.error(`Docker Compose file \`${composeFile}\` not found`)
+      process.exitCode = 1
+      return
+    }
+
+    // Create temporary directory to operate in
+    let tempdir = tmp.dirSync({ prefix: 'graph-test', unsafeCleanup: true }).name
+    try {
+      await configureTestEnvironment(toolbox, tempdir, composeFile, nodeImage)
+    } catch {
+      process.exitCode = 1
+      return
+    }
+
+    // Bring up test environment
+    try {
+      await startTestEnvironment(toolbox, tempdir)
+    } catch {
+      process.exitCode = 1
+      return
+    }
+
+    // Wait for test environment to come up
+    try {
+      await waitForTestEnvironment(toolbox, {
+        skipWaitForEthereum,
+        skipWaitForIpfs,
+        skipWaitForPostgres,
+      })
+    } catch {
+      await stopTestEnvironment(toolbox, tempdir)
+      process.exitCode = 1
+      return
+    }
+
+    // Bring up Graph Node separately, if a standalone node is used
+    let nodeProcess
+    let nodeOutputChunks = []
+    if (standaloneNode) {
+      try {
+        nodeProcess = await startGraphNode(
+          toolbox,
+          standaloneNode,
+          standaloneNodeArgs,
+          nodeOutputChunks,
+        )
+      } catch {
+        toolbox.print.error('')
+        toolbox.print.error('  Graph Node')
```

Comment:
I think we need better error messages here. I just got:

```
  Graph Node
  ----------

  Attaching to
```
And an exit. There are three instances of `Graph Node` with the line under it being printed, so it is also unclear which place this error is coming from (although in my case it  the fact that Graph Node could not be queried)

Reply:
I think the problem here was that docker compose was shut down before the output was collected. This should no longer be the case, I changed how this is done in the command.